### ISSUE TITLE
Off by one error?

### ISSUE THREAD

**Issue by keean** [GoogleChromeLabs/shadow-selection-polyfill#5]
It looks like there is an off by one error when there is a span that starts with a space. Something like the following in a shadow dom seems to cause the problem:
```html
<main contenteditable="true"><span>test</span><span> test</span></main>
```
If you try and select the second "test" word (double click on the second word in safari), it does not select the first character of the word.

**Comment by philcunliffe** [comment]
I'm having the exact same issue, any updates?

edit: did some investigation and it seems that if the first character is a space then line 379:

`offsetLeft = measure() + initialSpace(leftNode);  // measure doesn't include initial space`

The comment mentions that measure() does not include initial space but I'm seeing that it does. I'm not sure why it wouldn't.

edit2: Ok, this seems to come down collapsed and non-collapsed whitespace. As mentioned above, the measure() does catch non-collapsed whitespace and so the startOffset is incorrectly adjusted. The other major issue I found is that if there is a significant amount of collapsed whitespace to the right of the selection then the endOffset will also be off by the amount of collapsed whitespace

**Comment by samthor** [comment]
FWIW I believe this is because calling `toString()` on the `Selection` _intentionally_ returns the selection as the user sees it, i.e., with whitespace collapsed (if not `white-space: pre`).

This is a bit of a bummer, as it's not clear the the way to solve this. We could apply `white-space: pre` just for the code, so I'll see how costly that ends up being.

**Comment by samthor** [comment]
I've released a new version (1.1.0)—please try it out.

If you don't set `white-space: pre`, you might not be able to say, select initial whitespace (because you can't with your mouse or keyboard _anyway_). So it's impossible to select a range of e.g., 0-2 on a string like "  Hello, I start with two spaces".

But we should no longer barf or give invalid results :)
